src/s3path.jl

```diff
@@ -145,7 +152,7 @@ end
 
 function FilePathsBase.parents(fp::S3Path)
     if hasparent(fp)
-        return map(1:length(fp.segments)-1) do i
+        return map(0:length(fp.segments)-1) do i
```

Comment:
What's the reasoning for this change, just curious?

Reply:
Parents should include the root if we're dealing with an absolute path. In the case of S3Paths, they're always absolute paths, so this behaviour in FilePathsBase isn't necessary.

https://github.com/rofinn/FilePathsBase.jl/blob/master/src/path.jl#L224